controller/finalizer.go

```diff
@@ -84,18 +86,30 @@ func (f *Controller) addFinalizer(obj interface{}) (bool, error) {
 // removeFinalizer receives an object and tries to remove its finalizer which
 // was set by operatorkit. The removal of a finalizer will be retried and a fresh
 // object will get fetched from k8s if the ResourceVersion is out of date.
-func (f *Controller) removeFinalizer(obj interface{}) error {
+func (c *Controller) removeFinalizer(ctx context.Context, obj interface{}) error {
+	finalizerName := getFinalizerName(c.name)
+
+	c.logger.Log("level", "debug", "message", fmt.Sprintf("removing finalizer '%s' from runtime object", finalizerName))
+
+	if finalizerskeptcontext.IsKept(ctx) {
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not remove finalizer '%s' from runtime object", finalizerName))
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finalizer '%s' is requested to be kept in runtime object in this loop", finalizerName))
```

Comment:
I would say we drop `from runtime object` and be more brief. I find the long messages often rather not useful.  It is also that we log the object information in the form of its self link.

Reply:
Done.